storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -209,22 +224,34 @@ protected void initialize(Map conf, String localDir) throws Exception {
       throw new RuntimeException("Couldn't create generated-conf dir, _generatedConfPath=" + _generatedConfPath.toString());
     }
 
-    createLocalServerPort();
-    setupHttpServer();
-
-    _conf.put(Config.NIMBUS_HOST, _configUrl.getHost());
+    try {
+      mesosStormConf.put(Config.NIMBUS_HOST, MesosCommon.getNimbusHost(mesosStormConf));
+    } catch (UnknownHostException exp) {
+      LOG.error("Exception while configuring nimbus host: {}", exp);
+      // TODO(ksoundararaj): Should we exit here?
```

Comment:
Replaced lines 230 and 231 with `throw new RuntimeException(String.format("\"Exception while configuring nimbus host: %s", exp));`